### PR TITLE
RateMyApp update

### PR DESCRIPTION
**Pull request by alsvinicius**

### CODE REVIEW

File: src/main/java/com/mariosangiorgio/ratemyapp/RateMyApp.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
     public void appLaunched(){
         if(preferencesManager.alertEnabled()){
             if(canShowDialog()){
-                notificationManager.showDialog();
+                notificationManager.showDialog(1);
```
Comment (mariosangiorgio) [inline src/main/java/com/mariosangiorgio/ratemyapp/RateMyApp.java:39]: I think that using an enum would be clearer than with an integer. I'm probably going to change this with an enum or introducing a function for each phase.

File: src/main/java/com/mariosangiorgio/ratemyapp/RateMyApp.java [hunk 2]
```diff
@@ -70,15 +70,30 @@ private String getApplicationName(Context context){
...
-    public void showDialog() {
-        DialogListener listener = new DialogListener();
+    public void showDialog(int flowPosition) {
+
+        AlertDialog.Builder dialogBuilder =  new AlertDialog.Builder(context);
+        dialogBuilder.setTitle(context.getString(R.string.rate) + " " + getApplicationName(context));
+
+
+
+        switch(flowPosition){
```
Comment (mariosangiorgio) [inline src/main/java/com/mariosangiorgio/ratemyapp/RateMyApp.java:80]: See the other comment. Probably using two functions would make the code more readable.

File: src/main/java/com/mariosangiorgio/ratemyapp/RateMyApp.java [hunk 3]
```diff
@@ -101,4 +117,46 @@ public void onClick(DialogInterface dialogInterface, int buttonPressed) {
...
+        }
+
+        private class DialogListenerSendEmail implements DialogInterface.OnClickListener{
+
+            @Override
+            public void onClick(DialogInterface dialogInterface, int buttonPressed) {
+                switch(buttonPressed){
+                    case DialogInterface.BUTTON_POSITIVE:
+                        Intent emailIntent = new Intent(Intent.ACTION_SENDTO, Uri.fromParts(
+                                "mailto", "YOUR_EMAIL", null));
```
Comment (mariosangiorgio) [inline src/main/java/com/mariosangiorgio/ratemyapp/RateMyApp.java:150]: I'd rather let the user configure the e-mail through a parameter than having to replace this hard-coded value

**Comment by mariosangiorgio** [comment]
I was also thinking at leaving the possibility to choose between the single question and the "two phase question" you introduced. I think I'm going to use the latter but it would be nice to let everyone choose what fits the best for its application.

**Comment by alsvinicius** [comment]
I think that it's a fair option to the users that want to use the library but prefer the original implementation.

**Pull request merged**